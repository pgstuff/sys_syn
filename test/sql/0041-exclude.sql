BEGIN;

CREATE EXTENSION tinyint
    SCHEMA public;

CREATE EXTENSION sys_syn;

CREATE SCHEMA user_data
    AUTHORIZATION postgres;

CREATE TABLE user_data.test_table (
        test_table_key integer NOT NULL,
        test_table_text text,
        CONSTRAINT test_table_pkey PRIMARY KEY (test_table_key));

INSERT INTO sys_syn.in_groups_def VALUES ('in');

DO $$BEGIN
        EXECUTE sys_syn.in_table_add_sql('user_data.test_table'::regclass, 'in');
END$$;

INSERT INTO user_data.test_table(
        test_table_key, test_table_text)
VALUES (1,              'Exclude from in'),
       (2,              'Exclude from out'),
       (3,              'test_data');

INSERT INTO user_data.test_table_exclude VALUES (ROW(1));

INSERT INTO sys_syn.out_groups_def VALUES ('out');

SELECT sys_syn.out_table_add('user_data', 'test_table', 'out');

INSERT INTO user_data.test_table_out_exclude VALUES (ROW(2));

SELECT user_data.test_table_pull(FALSE);
SELECT user_data.test_table_out_move();

SELECT key, delta_type, queue_state FROM user_data.test_table_out_queue;

SELECT user_data.test_table_vacuum();

SELECT * FROM user_data.test_table_in;

ROLLBACK;
