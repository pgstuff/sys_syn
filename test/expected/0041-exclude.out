BEGIN;
CREATE EXTENSION tinyint
    SCHEMA public;
CREATE EXTENSION sys_syn;
CREATE SCHEMA user_data
    AUTHORIZATION postgres;
CREATE TABLE user_data.test_table (
        test_table_id integer NOT NULL,
        test_table_text text,
        CONSTRAINT test_table_pid PRIMARY KEY (test_table_id));
INSERT INTO sys_syn.in_groups_def VALUES ('in');
DO $$BEGIN
        EXECUTE sys_syn.in_table_add_sql('user_data.test_table'::regclass, 'in');
END$$;
INSERT INTO user_data.test_table(
        test_table_id, test_table_text)
VALUES (1,              'Exclude from in'),
       (2,              'Exclude from out'),
       (3,              'test_data');
INSERT INTO user_data.test_table_exclude VALUES (ROW(1));
INSERT INTO sys_syn.out_groups_def VALUES ('out');
SELECT sys_syn.out_table_add('user_data', 'test_table', 'out');
 out_table_add 
---------------
 
(1 row)

INSERT INTO user_data.test_table_out_exclude VALUES (ROW(2));
SELECT user_data.test_table_pull(FALSE);
 test_table_pull 
-----------------
 t
(1 row)

SELECT user_data.test_table_out_move();
 test_table_out_move 
---------------------
 t
(1 row)

SELECT id, delta_type, queue_state FROM user_data.test_table_out_queue;
 id  | delta_type | queue_state 
-----+------------+-------------
 (3) | Add        | Unclaimed
(1 row)

SELECT user_data.test_table_vacuum();
 test_table_vacuum 
-------------------
 
(1 row)

SELECT * FROM user_data.test_table_in;
 trans_id_in | id  | attributes  | no_diff 
-------------+-----+-------------+---------
           1 | (3) | (test_data) | 
(1 row)

ROLLBACK;
